src/modules/user/apiKeys/sagas/apiKeyCreateSaga.ts

```diff
@@ -16,10 +15,17 @@ export function* apiKeyCreateSaga(action: ApiKeyCreateFetch) {
     try {
         const apiKey = yield call(API.post(createOptions(getCsrfToken())), '/resource/api_keys', action.payload);
         yield put(apiKeyCreate(apiKey));
-        yield put(alertPush({message: ['success.api_keys.created'], type: 'success'}));
-        yield put(apiKeys2FAModal({active: true, action: 'createSuccess', apiKey}));
+        yield put(alertPush({ message: ['success.api_keys.created'], type: 'success' }));
+        yield put(apiKeys2FAModal({ active: true, action: 'createSuccess', apiKey }));
     } catch (error) {
-        yield put(alertPush({message: error.message, code: error.code, type: 'error'}));
-        yield put(apiKeys2FAModal({active: false}));
+        yield put(sendError({
+            error,
+            processingType: 'alert',
+            extraOptions: {
+                type: action.type,
+                actionError: apiKeysError,
+                apiKeys2FAModal,
+            },
+        }));
```

Comment:
this should work:

        yield put(apiKeys2FAModal({ active: true, action: 'createSuccess', apiKey }));
        yield put(apiKeysError(error));  
        yield put(sendError({
            error,
            processingType: 'alert',
        }));